src/EnergyPlus/HeatRecovery.cc

```diff
@@ -1501,7 +1502,10 @@ namespace HeatRecovery {
                 }
             }
 
-            if (CompanionCoilIndex > 0) {
+            if ((((CompanionCoilType_Num == DataHVACGlobals::CoilDX_CoolingSingleSpeed) ||
+                  (CompanionCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed)) &&
+                 (CompanionCoilIndex > 0)) ||
+                ((CompanionCoilType_Num == DataHVACGlobals::CoilDX_Cooling) && (CompanionCoilIndex > -1))) {
```

Comment:
@jcyuan2020 this check probably isn't even needed. Yes, the new coil model required a change here but there must be a companion coil and it must be valid since it was verified in getInput. Open a new issue to remove these types of checks in this model.